### PR TITLE
Display versions of CUDA libraries

### PR DESCRIPTION
**Pull request by kmaehashi**
Ref #2569.

```py
>>> cupy.show_config()
CuPy Version          : 7.0.0rc1
CUDA Root             : /usr/local/cuda
CUDA Build Version    : 9020
CUDA Driver Version   : 9020
CUDA Runtime Version  : 9020
cuBLAS Version        : 9020
cuFFT Version         : 9020
cuRAND Version        : 9020
cuSOLVER Version      : (9, 2, 0)
cuSPARSE Version      : 9020
NVRTC Version         : (9, 2)
cuDNN Build Version   : 7500
cuDNN Version         : 7500
NCCL Build Version    : 2307
NCCL Runtime Version  : 2307
```

### CODE REVIEW

**Comment by niboshi** [comment]
Let's get rid of `details` argument and always print all the information.

**Comment by leofang** [comment]
Related? #756

**Comment by kmaehashi** [comment]
> Related? #756
Ah, this was intended to be closed in the initial impl of `show_config`.

**Comment by kmaehashi** [comment]
> Let's get rid of details argument and always print all the information.
Removed it.

For later reference: in my environment automatically-created handles consume additional 56 MiB (on GTX TITAN X), but we think its acceptable as `show_config()` is mainly used for diagnosis purpose.

**Comment by jakirkham** [comment]
This would be really great to have. Thanks for working on it @kmaehashi! 😄

File: cupy/cuda/cufft.pyx [hunk 1]
```diff
@@ -57,6 +57,9 @@ cdef extern from 'cupy_cufft.h' nogil:
     Result cufftExecD2Z(Handle plan, Double *idata, DoubleComplex *odata)
     Result cufftExecZ2D(Handle plan, DoubleComplex *idata, Double *odata)
 
+    # Version
+    Result cufftGetVersion(int *version)
```
Comment (niboshi) [inline cupy/cuda/cufft.pyx]:
```suggestion
    Result cufftGetVersion(int* version)
```

**Comment by niboshi** [comment]
In my environment, cupy becomes unable to import.
```
$ python -c 'import cupy'
...
ImportError: CuPy is not correctly installed.

If you are using wheel distribution (cupy-cudaXX), make sure that the version of CuPy you installed matches with the version of CUDA on your host.
Also, confirm that only one CuPy package is installed:
  $ pip freeze

If you are building CuPy from source, please check your environment, uninstall CuPy and reinstall it with:
  $ pip install cupy --no-cache-dir -vvvv

Check the Installation Guide for details:
  https://docs-cupy.chainer.org/en/latest/install.html

original error: /data2/work/w/repos/cupy2/cupy/cuda/cusparse.cpython-38-x86_64-linux-gnu.so: undefined symbol: cusparseSbsric02_bufferSizeExt
```

File: cupy/cuda/cufft.pyx (hunk 1 above)
Comment (kmaehashi) [reply cupy/cuda/cufft.pyx]: It's a style of CUDA (I copied this from header file) to use `int*`... but I agree with your proposal

**Comment by kmaehashi** [comment]
@niboshi I think the build issue is unrelated to this PR (it is one of a symbol removed in CUDA 10.1 Update 2), so could you try rebuilding?

**Comment by niboshi** [comment]
@kmaehashi Sorry it looks like so. It happens in master (acd7010bf225dab2d0e5fb8306c8988e822af300) too.

File: cupy/cuda/cupy_cuda_common.h [hunk 2]
```diff
@@ -154,6 +154,18 @@ struct cudaTextureDesc {
 };
 
 
+///////////////////////////////////////////////////////////////////////////////
+// library_types.h
+///////////////////////////////////////////////////////////////////////////////
+
+typedef enum libraryPropertyType_t
+{
```
Comment (niboshi) [inline cupy/cuda/cupy_cuda_common.h]: This one should be appended to the previous line.

File: cupy/cuda/cusolver.pyx [hunk 3]
```diff
@@ -29,6 +29,14 @@ cdef extern from 'cupy_cusolver.h' nogil:
     int cusolverDnSetStream(Handle handle, driver.Stream streamId)
     int cusolverSpSetStream(SpHandle handle, driver.Stream streamId)
 
+    # Library Property
+    int cusolverGetProperty(LibraryPropertyType type, int *value)
```
Comment (niboshi) [inline cupy/cuda/cusolver.pyx]: Ditto. as the previous review
```suggestion
    int cusolverGetProperty(LibraryPropertyType type, int* value)
```

**Comment by kmaehashi** [comment]
Thanks, fixed as suggested.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**